azurerm/resource_arm_devspace_controller.go

```diff
@@ -0,0 +1,271 @@
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroupName := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	sku := expandDevSpaceControllerSku(d)
+
+	hostSuffix := d.Get("host_suffix").(string)
+	tarCHResId := d.Get("target_container_host_resource_id").(string)
+	tarCHCredBase64 := d.Get("target_container_host_credentials_base64").(string)
+
+	controller := devspaces.Controller{
+		Location: &location,
+		Tags:     expandTags(tags),
+		Sku:      sku,
+		ControllerProperties: &devspaces.ControllerProperties{
+			HostSuffix:                           &hostSuffix,
+			TargetContainerHostResourceID:        &tarCHResId,
+			TargetContainerHostCredentialsBase64: &tarCHCredBase64,
+		},
+	}
+
+	future, err := client.Create(ctx, resGroupName, name, controller)
+	if err != nil {
+		return fmt.Errorf("Error creating DevSpace Controller %q (Resource Group %q): %+v", name, resGroupName, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of DevSpace Controller %q (Resource Group %q): %+v", name, resGroupName, err)
+	}
+
+	result, err := client.Get(ctx, resGroupName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving DevSpace %q (Resource Group %q): %+v", name, resGroupName, err)
+	}
+
+	if result.ID == nil {
+		return fmt.Errorf("Cannot read DevSpace Controller %q (Resource Group %q) ID", name, resGroupName)
+	}
+	d.SetId(*result.ID)
+
+	return resourceArmDevSpaceControllerRead(d, meta)
+}
+
+func resourceArmDevSpaceControllerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).devSpaceControllerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroupName := id.ResourceGroup
+	name := id.Path["controllers"]
+
+	result, err := client.Get(ctx, resGroupName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(result.Response) {
+			log.Printf("[DEBUG] DevSpace Controller %q was not found in Resource Group %q - removing from state!", name, resGroupName)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on DevSpace Controller Lab %q (Resource Group %q): %+v", name, resGroupName, err)
+	}
+
+	d.Set("name", result.Name)
+	d.Set("resource_group_name", resGroupName)
+	if location := result.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	d.Set("sku", flattenDevSpaceControllerSku(result.Sku))
+
+	if props := result.ControllerProperties; props != nil {
+		if props.HostSuffix != nil {
+			d.Set("host_suffix", props.HostSuffix)
+		}
+
+		if props.DataPlaneFqdn != nil {
+			d.Set("data_plane_fqdn", props.DataPlaneFqdn)
+		}
+
+		if props.TargetContainerHostResourceID != nil {
+			d.Set("target_container_host_resource_id", props.TargetContainerHostResourceID)
+		}
```

Comment:
we should set these properties even if they're nil (which would be set as an empty string) so the diff is displayed